code/computational_geometry/src/sphere_tetrahedron_intersection/sphere_tetrahedron_intersection.cpp

```diff
@@ -0,0 +1,1237 @@
+{
+	// LuVector.hpp
+	
+	using namespace std;
+
+	template< size_t N, class T >
+	class LuVector;
+
+	template< class T >
+	using LuVector2 = LuVector< 2, T >;
+
+	template< class T >
+	using LuVector3 = LuVector< 3, T >;
+
+	using LuVector2f = LuVector< 2, float >;
+	using LuVector2d = LuVector< 2, double >;
+	using LuVector2c = LuVector< 2, complex< double > >;
+
+	using LuVector3f = LuVector< 3, float >;
+	using LuVector3d = LuVector< 3, double >;
+	using LuVector3c = LuVector< 3, complex< double > >;
+	
+	// LuVector_Op.hpp
+	
+	//////////////////////////////////////////////////////////////
+	// TEMPLATE LOOP INDEX VARIABLE
+	//////////////////////////////////////////////////////////////
+	template< size_t N >
+	struct LOOP_INDEX
+	{
+
+	};
+
+	//////////////////////////////////////////////////////////////
+	// TEMPLATE LOOP UNROLLING OPERATION FUNCTORS
+	//////////////////////////////////////////////////////////////
+
+	template< class T, class S >
+	struct OP_COPY
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = rhs;
+		}
+	};
+
+	//////////////////////////////////////////////////////////////
+
+	template< class T, class S >
+	struct OP_ADD
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs += rhs;
+		}
+	};
+
+	template< class T, class S >
+	struct OP_SUB
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs -= rhs;
+		}
+	};
+
+	template< class T, class S >
+	struct OP_MUL
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs *= rhs;
+		}
+	};
+
+	template< class T, class S >
+	struct OP_DIV
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs /= rhs;
+		}
+	};
+
+	//////////////////////////////////////////////////////////////
+
+	template< class T, class S >
+	struct OP_ABS
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = abs( rhs );
+		}
+	};
+
+	template< class T, class S >
+	struct OP_ARG
+	{
+		inline void operator ()( T& lhs, const S& rhs )
+		{
+			lhs = arg( rhs );
+		}
+	};
+
+	template< size_t N, class T, class S >
+	struct OP_MIN
+	{
+		inline void operator ()( LuVector< N, T >& result, const LuVector< N, T >& lhs, const LuVector< N, S >& rhs, const size_t& I )
+		{
+			result[ I ] = lhs[ I ] < rhs[ I ] ? lhs[ I ] : rhs[ I ];
+		}
+	};
+
+	template< size_t N, class T, class S >
+	struct OP_MAX
+	{
+		inline void operator ()( LuVector< N, T >& result, const LuVector< N, T >& lhs, const LuVector< N, S >& rhs, const size_t& I )
+		{
+			result[ I ] = lhs[ I ] > rhs[ I ] ? lhs[ I ] : rhs[ I ];
+		}
+	};
+
+	template< size_t N, class T, class S >
+	struct OP_DOT
+	{
+		inline void operator ()( T& result, const LuVector< N, T >& lhs, const LuVector< N, S >& rhs, const size_t& I )
+		{
+			result += lhs[ I ] * rhs[ I ];
+		}
+	};
+
+	//////////////////////////////////////////////////////////////
+
+	template< class T >
+	struct OP_OSTREAM
+	{
+		inline void operator ()( ostream& lhs, const T& rhs, const string& delimeter )
```

Comment:
Minor typo, but I think this isn't the correct spelling of delimiter.

Reply:
Ah, thanks for pointing out. No, not a typo, I knew it as 'delimeter' :) English is not my main language and I have never noticed that it was actually 'delimiter'.